backend/src/services/calendar/api/methods.py

```
@@ -105,9 +105,70 @@
+        try:
+            dt = parse_rfc3339(expiration)
+            # Convert to milliseconds since epoch
+            return int(dt.timestamp() * 1000)
+        except (ValueError, AttributeError):
+            pass
+
+        # Invalid format
+        raise ValidationError(
+            f"Invalid expiration format: {expiration}. "
+            "Expected milliseconds since epoch or ISO 8601 date string."
+        )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)